language/move-lang/src/expansion/ast.rs

```diff
@@ -445,31 +444,29 @@ impl Address {
 
     pub fn into_addr_bytes_opt(
         self,
-        addresses: &UniqueMap<Name, AddressBytes>,
+        addresses: &BTreeMap<String, AddressBytes>,
     ) -> Option<AddressBytes> {
         match self {
             Self::Anonymous(sp!(_, bytes)) => Some(bytes),
-            Self::Named(n) => addresses.get(&n).cloned(),
+            Self::Named(n) => addresses.get(&n.value).cloned(),
         }
     }
 
     pub fn into_addr_bytes(
         self,
-        addresses: &UniqueMap<Name, AddressBytes>,
+        addresses: &BTreeMap<String, AddressBytes>,
         loc: Loc,
         case: &str,
     ) -> Result<AddressBytes, Diagnostic> {
```

Comment:
You should be able to remove the errors here I think, given the other errors. You might need to wait for #8881 to land though

Reply:
I'm probably misunderstanding something here, but if the `UnboundAddress` is a non-blocking error don't we need to keep this error? Since we can possibly encounter a named address without a value later on and will need to raise a `BlockingError` at that point?

Reply:
I'm saying swapping `UnboundAddress` from `BlockingError` to `NonblockingError`, perhaps the name is confusing here but the `Nonblocking` bit is that it doesn't block other compiler passes. It still blocks bytecode generation

Reply:
Thanks for the clarification -- and realized what you're saying here when I was reviewing that PR. Should have these changes up in a couple minutes.

Reply:
All removed, and rebased on top of that PR and it all works 😃